prometheus/counter.go

```diff
@@ -121,6 +123,10 @@ func (c *counter) Add(v float64) {
 		panic(errors.New("counter cannot decrease in value"))
 	}
 
+	if c.createdTs == nil {
+		c.createdTs = timestamppb.Now()
```

Comment:
Do we maybe want a `c.timestampPbNow` in a similar fasion to `c.now` for testing?

Reply:
Oh, now I get how this could be tested! I was so confused how I could create a deterministic test with something that depends on machine clock 😭